### DISCUSSION TITLE
Cannot override `index=True` with mapped_column in Annotated

### DISCUSSION THREAD

**Discussion by decardev** [sqlalchemy/sqlalchemy#11090] [Usage Questions] [answered]
Cannot override `mapped_column(index=True)` when using `Annotated`

`python` 3.12
`OK`: OSX
`sqlalchemy`: 2.0.27

The [documentation](https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#mapping-whole-column-declarations-to-python-types) states:

> The [mapped_column()](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.mapped_column) construct is also reconciled with an explicitly passed [mapped_column()](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.mapped_column) construct, whose arguments will take precedence over those of the Annotated construct. Below we add a [ForeignKey](https://docs.sqlalchemy.org/en/20/core/constraints.html#sqlalchemy.schema.ForeignKey) constraint to our integer primary key and also use an alternate server default for the created_at column:

But the following produces an indexe for `fail`.

```python
from typing import Annotated

from pydantic.dataclasses import dataclass
from sqlalchemy import Integer, inspect
from sqlalchemy.orm import DeclarativeBase, Mapped, MappedAsDataclass, mapped_column


class Base(MappedAsDataclass, DeclarativeBase, dataclass_callable=dataclass):
    pass


IdField = Annotated[int, mapped_column(Integer, index=False)]
IdFieldIndexed = Annotated[int, mapped_column(Integer, index=True)]


class Model(Base):
    __tablename__ = "model"
    id: Mapped[int] = mapped_column(Integer, primary_key=True, default=1)
    ok_default: Mapped[IdField] = mapped_column(default=1)
    ok_overrides_false: Mapped[IdField] = mapped_column(index=True, default=1)

    fail: Mapped[IdFieldIndexed] = mapped_column(index=False, default=1)


# mapped_column does not override the index parameter of the Annotated type when True. 
{name: col.expression.index for name, col in inspect(Model).column_attrs.items()}
```

**Comment by zzzeek** [comment, marked as answer]
thanks for reporting, a patch is in review at #11091

**Reply by decardev** [reply]
Thank you!
